docs: Improve CAS to Karpenter migration steps for clarity and automation

mozammil89 opened the pull request:
<!-- Please follow the guidelines at https://www.conventionalcommits.org/en/v1.0.0/ and use one of the following in your title:
feat:            <-- New features that require a MINOR version update
fix:             <-- Bug fixes that require at PATCH version update
chore:           <-- Smaller changes that impact behavior but aren't large enough to be features
perf:            <-- Code changes that improve performance but do not impact behavior
docs:            <-- Documentation changes that do not impact code
test:            <-- Test changes that do not impact behavior
ci:              <-- Changes that affect test or rollout automation
!${type}:        <-- Include ! if your change includes a backwards incompatible change.

Please review the Karpenter contribution docs at https://karpenter.sh/docs/contributing/ before submitting your pull request.
-->

Fixes #N/A <!-- issue number -->

**Description**
- Added CloudFormation template to automate setup of IAM role, policy, and interruption queue for EC2 events.
- Refined security group tagging, separating SG configurations for Launch Templates for better clarity.
- Simplified Karpenter deployment by passing node affinity settings directly to the Helm chart, eliminating the need for manual CRD installation.
- Made minor updates to improve overall readability and clarity in the documentation.

**How was this change tested?**
- Doc updates tested locally with hugo server
- Dry-run of steps in an existing EKS cluster with managed node group

**Does this change impact docs?**
- [x] Yes, PR includes docs updates <!-- docs must be added to /preview to be included in future version releases -->
- [ ] Yes, issue opened: # <!-- issue number -->
- [] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

mariuskimmina commented on website/content/en/preview/getting-started/migrating-from-cas/_index.md:
```diff
@@ -10,65 +10,91 @@ This guide will show you how to switch from the [Kubernetes Cluster Autoscaler](
-* You will use existing VPC and subnets
-* You will use existing security groups
+* You will use an existing VPC and subnets
+* You will use an existing security groups
```
Should not have an `an` because security groups is plural

mariuskimmina commented on website/content/en/preview/getting-started/migrating-from-cas/_index.md:
```diff
@@ -10,65 +10,91 @@ This guide will show you how to switch from the [Kubernetes Cluster Autoscaler](
 * Your cluster has an [OIDC provider](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html) for service accounts
 
-This guide will also assume you have the `aws` CLI installed.
+This guide will also assume you have the `aws` CLI and Helm client installed.
```
Just `helm` instead Helm client?

mariuskimmina commented on website/content/en/preview/getting-started/migrating-from-cas/_index.md:
```diff
@@ -88,53 +114,54 @@ The full aws-auth configmap should have two groups.
-          operator: In
-          values:
-          - ${NODEGROUP}
-  podAntiAffinity:
-    requiredDuringSchedulingIgnoredDuringExecution:
-      - topologyKey: "kubernetes.io/hostname"
```
I believe this should stay, why would we remove it?

mariuskimmina replied:
Still wondering why you want to remove the `podAntiAffinity` here

mozammil89 commented:
@mariuskimmina Thank you for reviewing PR. I've addressed your review comments. Could you please recheck and let me know if further feedback is required? If everything looks good, kindly approve the PR.

mariuskimmina commented on website/content/en/preview/getting-started/migrating-from-cas/_index.md:
```diff
@@ -16,59 +16,85 @@ We will make the following assumptions in this guide
 {{% script file="./content/en/{VERSION}/getting-started/migrating-from-cas/scripts/step01-env.sh" language="bash" %}}
 
-Use that information to create our IAM roles, inline policy, and trust relationship.
+{{% alert title="Warning" color="warning" %}}
+If you open a new shell to run steps in this procedure, you need to set the environment variables again.
+{{% /alert %}}
```
Do we really expect every piece of documentation ever that uses env variables to explain / mention this? If we add it here it kinda implies it should be added everywhere that env variables are mentioned.

mariuskimmina commented on website/content/en/preview/getting-started/migrating-from-cas/_index.md:
```diff
@@ -16,59 +16,85 @@ We will make the following assumptions in this guide
 * Your cluster has an [OIDC provider](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html) for service accounts
 
-This guide will also assume you have the `aws` CLI installed.
+This guide will also assume you have the `aws` CLI and Helm installed.
```
should be `helm` not Helm

mariuskimmina commented on website/content/en/preview/getting-started/migrating-from-cas/_index.md:
```diff
@@ -16,59 +16,85 @@ We will make the following assumptions in this guide
-To create the Karpenter node role we will use the following policy and commands.
-
-{{% script file="./content/en/{VERSION}/getting-started/migrating-from-cas/scripts/step02-node-iam.sh" language="bash" %}}
+Use CloudFormation to set up the infrastructure needed by the existing EKS cluster. See [CloudFormation]({{< relref "../../reference/cloudformation/" >}}) for a complete description of what `cloudformation.yaml` does for Karpenter. The provided `cloudformation.yaml` template simplifies this setup by creating and configuring all necessary resources, including:
```
Personally, I prefer the use of `aws` commands over cloudformation, as I'd end up doing it all with terraform anyway but not sure what other things or what the general opinion on this is.
That said, you have a point here in that the cloudformation will script will also setup the interruption queue and policies, which are later used but never setup in the current version of this document.

mozammil89 replied:
That's correct.

mozammil89 reviewed (commented):
I've addressed all the review comments.

mariuskimmina reviewed (approved):
Seems like a good improvement to me, also left a message in Slack trying to get more eyes on it

jonathan-innis commented:
Assigning this to @njtran to figure out how we keep this data up-to-date as we make changes to our base Getting Started Guide. I think the challenge here is that we often make changes to one and not the other.

mozammil89 commented:
> Assigning this to @njtran to figure out how we keep this data up-to-date as we make changes to our base Getting Started Guide. I think the challenge here is that we often make changes to one and not the other.

The majority of customers I work with already have existing EKS clusters and naturally follow the CAS to Karpenter migration path. You’re absolutely right that both guides need to be updated in parallel. Please feel free to reach out to me on Slack at @mozkhan if you need any additional information for this PR.

rafael-debug commented:
Very good

jonathan-innis commented:
Unassigning @njtran since he's moved on from AWS and is expected to be less active in the community. @njtran please let us know if you want to be reassigned 😃

mozammil89 commented:
> Unassigning @njtran since he's moved on from AWS and is expected to be less active in the community. @njtran please let us know if you want to be reassigned 😃

While we're checking with @njtran, could you please check if any other active members would be interested to pick this up?

jonathan-innis commented:
@mozammil89 Can you go ahead and rebase this PR? Looks like we fell behind on this one but we're working on getting assignee if we can get the PR up-to-date with the latest main!

jonathan-innis commented:
Closing since it's been 30d since the last response
